.github/copilot-instructions.md

```diff
@@ -0,0 +1,157 @@
+  - Requires YouTube API credentials
+- **Get next Sunday date**: `npm run get-next-sunday`
+  - Utility script for date calculation
+  - Used by other automation scripts
+
+### CodeceptJS Testing
+
+- Run website automation tests: `npm run codeceptjs`
+- Run headless tests: `npm run codeceptjs:headless`
+- Configuration: `src/website_update/codecept.conf.js`
+- Tests target WordPress admin interface at lkg-halle.de
+
+## Validation
+
+### Before Making Changes
+
+- **ALWAYS run formatting check**: `npx prettier --check .`
+- **ALWAYS run formatting fix before committing**: `npx prettier --write .`
+- **Test website build**: `cd website && npm run build`
+- **Test website dev server**: `cd website && npm run dev`
+
+### After Making Changes
+
+- **Manual validation scenarios**:
+  1. **Website functionality**: Start dev server, verify pages load at localhost:4321
+  2. **Build process**: Ensure `npm run build` completes without errors
+  3. **Script syntax**: Test `npm run get-next-sunday` (will show auth error but proves syntax is correct)
+  4. **Formatting**: Run `npx prettier --check .` and fix any issues
+
+### CI/CD Integration
+
+- GitHub Actions automatically run on schedule (Sundays 18:00 UTC)
+- Actions require all secrets configured in repository settings
+- Dependabot updates dependencies monthly with grouped minor/patch updates
+
+## Project Structure
+
+### Root Directory Scripts
+
+- `/src/calendar_update/` - Google Calendar API integration
+- `/src/website_update/` - WordPress automation with CodeceptJS
+- `/src/youtube_update/` - YouTube API integration
+- `/src/csv-parser/` - CSV file processing utilities
+- `/input-csvs/` - Event planning CSV files
+
+### Website Directory
+
+- `/website/src/pages/` - Astro page components
+- `/website/src/layouts/` - Astro layout components
+- `/website/public/` - Static assets
+- `/website/dist/` - Built website output (generated)
+
+## Common Development Tasks
+
+### Repository Root Files
+
+```bash
+ls -la
+# .env
+# .nvmrc (Node 24.5)
+# README.md
+# package.json
+# src/
+# website/
+# input-csvs/
+# todo.md
+```
+
+### Key Dependencies
+
+- **codeceptjs**: Browser automation framework (locked at 3.7.3)
+- **playwright**: Browser driver for CodeceptJS
+- **googleapis**: Google Calendar and YouTube API client
+- **dotenv**: Environment variable management
+- **dayjs**: Date manipulation utilities
+
+### Environment Considerations
+
+- **Node version**: Specified in .nvmrc as 24.5, works with 20.19.4+
+- **Playwright browsers**: Required for automation, install with `npx playwright install --with-deps`
+- **Google API**: Service account required for calendar access
+- **WordPress credentials**: Admin access required for slider updates
+
+## Troubleshooting Common Issues
+
+### Build Failures
+
+- **Prettier formatting**: Run `npx prettier --write .` to fix formatting issues
+- **Missing dependencies**: Run `npm install` in both root and website directories
```

Comment:
Please use `npm ci` instead. We don't want the lockfile to change unnecessariy. Only use install if you need to install new dependencies